nats/js/api.py

```diff
@@ -276,6 +276,7 @@ class StreamConfig(Base):
     max_msgs: Optional[int] = None
     max_bytes: Optional[int] = None
     discard: Optional[DiscardPolicy] = DiscardPolicy.OLD
+    discard_new_per_subject: bool = False
```

Comment:
What would proper coverage look like?

I do see places where we're verifying that defaults are correctly applied -- so one might add a line like `        assert config.discard_new_per_subject == False` -- but it's a bit less obvious how to test past there, _particularly_ since the documentation for the server side doesn't describe exactly how things misbehave when `DiscardPolicy.NEW` is in use and `MaxMsgsPerSubject` is set but `DiscardNewPerSubject` isn't also value, but merely specifies that the value _should_ be used when combining those two values.